docs/02_database/01_fields-tables-views/02_tables/01_tables-basics.md

```diff
@@ -100,38 +105,113 @@ tables {
+
+- A **unique index** ensures that no two records in the table can have the same value for the specified field or fields. For example, this could be a TRADE_ID in a TRADES table, where the unique value is generated via autoIncrement or sequence. Or it could be a CURRENCY_SYMBOL in a table of CURRENCIES.  
+- A **non-unique index** is useful when uniqueness is not important or not possible. For example, you could add a non-unique index on the ORDER_ID field in your TRADE table. This enables you to find all the trades that match a specified order.  (ORDER_ID would be unique in the ORDERS table, but the order can be filled by multiple trades.) 
+
+Here is a simple example showing a unique and a non-unique index.  Each index has only one field.
+
+```
+  table(name = "TRADE", id = 1010) {
+      autoIncrement(TRADE_ID)
+      INSTRUMENT_ID
+      TRADE_TYPE
+      TRADE_DATE
+      CURRENCY_ID
+      QUANTITY
+      PRICE
+      
+      primaryKey {
+          TRADE_ID 
+      }
+      indices {
+          unique {
+                   INSTRUMENT_ID
+          }
+          nonUnique {
+                   CURRENCY_ID
+          }
+      }
+ }
+
+}
+```
+
+See our page on [operations and indices](../../../../database/data-structures/indices/) for information on the operations that can be used to perform look-ups on an index.
 
-- Creating additional unique constraints to the table's primary key
-- Providing a performant record lookup (`unique` index)
-- Providing an index for a performant ranged lookup (`nonUnique` index, else `unique` index but searching on part of the key)
+### Enforcing uniqueness
+We have already noted that a unique index enforces uniqueness on that field. So you can add a unique index to a table to create a further constraint on the primary index.
 
-You can define indices on a table in a similar fashion to `primaryKey`. As with keys, indices are made up of one or more fields from the table. Where the indices are made up of multiple fields, the order of the fields matters if you want to search based on a partial index. This affects performance.
+Imagine a scenario where you have projects and users.
+
+- You have a PROJECTS table where each project has a unique PROJ_ID.
+- You have a USERS table where each project has a unique USER_ID.
+
+There is a many-to-many relationship between the two tables - projects can have more than one user, and users can belong to more than one project. To allocate users to projects, you need a third table: PROJECT_USERS.
+
+In the the PROJECT_USERS table, you only need two fields:
+
+- PROJ_ID
+- USER_ID
+
+Each record in this table states that PROJ_ID= x has USER_ID=y. But it is important that you do not duplicate any of these records - which could cause problems. So, in your PROJECT_USERS table, make PROJ_ID the primary index and add a unique index on PROJ_ID and USER_ID. The database will now prevent you from entering a record that has exactly the same PROJ_ID and USER_ID as an existing record.
```

Comment:
I would rewrite this along these lines: 

Each record in this table states that PROJ_ID = x has USER_ID = y. However we have a requirement that any combination of PROJ_ID and USER_ID only occurs once in our table. So in our table, we define the primary key on PROJ_ID and USER_ID. This means we can rely on the database to ensure uniqueness. Inserting a record with the same PROJ_ID and USER_ID that already exist will fail.